commons/src/config/mod.rs

```diff
@@ -0,0 +1,192 @@
+use std::{collections::HashSet, path::PathBuf};
+
+use clap::{crate_authors, Args, Parser};
+use logid::{log_id::LogId, set_event_with};
+use serde::{Deserialize, Serialize};
+
+use crate::log_id::COMMONS_LOG_ID_MAP;
+
+use self::{log_id::ConfigErrLogId, preamble::Preamble};
+
+pub mod log_id;
+pub mod output;
+pub mod preamble;
+
+const UNIMARKUP_NAME: &str = "unimarkup";
+const ABOUT: &str = "The official compiler for Unimarkup.";
+const HELP_TEMPLATE: &str = r#"
+{before-help}{name} {version} - {about-with-newline}
+Written by: {author-with-newline}
+{usage-heading} {usage}
+
+{all-args}{after-help}"#;
+
+/// Trait defining functions every configuration struct must implement.
+pub trait ConfigFns {
+    /// Merges another configuration struct into **self**.
+    /// Merging is done according to the [Unimarkup specification](https://github.com/unimarkup/specification/blob/main/configuration/merging-configurations.md).
+    ///
+    /// ## Arguments
+    ///
+    /// * `other` ... The configuration struct that is consumed and merged into **self**
+    fn merge(&mut self, other: Self);
+
+    /// Checks if all set values are valid.
+    /// e.g. that set files exist
+    fn validate(&self) -> Result<(), LogId>;
+
+    /// Returns `true` if `validate()` returned `Ok`.
+    fn is_valid(&self) -> bool {
+        Self::validate(self).is_ok()
```

Comment:
I'm curious what would happen if some type implemented this function twice, one as it's own function and one when implementing this trait. I guess here the `ConfigFns` function should be executed, but I am really not sure haha. 

Disambiguation here is possible (I think):

```suggestion
        ConfigFns::validate(self).is_ok()
```